libphx/src/shader.rs

```diff
@@ -267,16 +290,21 @@ pub unsafe extern "C" fn Shader_ToShaderState(this: &mut Shader) -> Box<ShaderSt
 #[no_mangle]
 pub unsafe extern "C" fn Shader_Start(this: &mut Shader) {
     Profiler_Begin(c_str!("Shader_Start"));
+
     gl::UseProgram(this.program);
+
     current = this;
     this.texIndex = 1;
 
     /* Fetch & bind automatic variables from the shader var stack. */
     let mut i: i32 = 0;
     while i < this.vars.len() as i32 {
         let var: &mut ShaderVar = &mut this.vars[i as usize];
+
         if !((*var).index < 0) {
-            let pValue: *mut libc::c_void = ShaderVar_Get((*var).name.as_ptr(), (*var).type_0);
+            let c_name = static_string!((*var).name.as_str());
```

Comment:
Why are we pinning the lifetime of `c_name` here? It's passed to `ShaderVar_Get`, so the lifetime is already greater than the duration that `ShaderVar_Get` needs it, even if it's a local variable. If `ShaderVar_Get` needs to store that string for any longer, it can copy it somewhere else.

Reply:
Reverted

Reply:
Let's leave it for now, until we make `ShaderVar_Get` better.